### DISCUSSION TITLE
VisualFBEditor 1.2.9

### DISCUSSION THREAD

**Discussion by XusinboyBekchanov** [XusinboyBekchanov/VisualFBEditor#83] [General]
Version 1.2.9 (July 14, 2021)
- Fixed: Show windows on DPI sets
- Added: Polish language
- Fixed: Showing lines in Code Editor
- Fixed: IDE on exists command line not shows New Project dialog
- Fixed: COMWrapperBuilder
- Fixed: mff dll path on Load Toolbox
- Fixed: Open Command Prompt, if main file is empty

<hr /><em>This discussion was created from the release <a href='https://github.com/XusinboyBekchanov/VisualFBEditor/releases/tag/1.2.9'>VisualFBEditor 1.2.9</a>.</em>

**Comment by AntonioFS-1962** [comment]
Greetings from Sabadell-Barcelona-Spain.

First of all, lots of encouragement to continue developing this very interesting IDE for our beloved FreeBasic. Congratulations for such a great job. :-)

I must mention that in version 1.2.9 for Linux (in my case Mint 20), there is an error when running the program:
File not found -> libmff64_gtk3.so
Can not load control to toolbox

I will try with an older version, to see if the error mentioned comes from behind.

Thank you very much.
Antonio F.S.

**Reply by XusinboyBekchanov** [reply]
Try compiling from sources on github.

Checked with this function:
`If g_file_test (ToUTF8 (* filename.vptr), G_FILE_TEST_EXISTS) Then`

Does the g_file_test function not work on your system?

**Reply by XusinboyBekchanov** [reply]
You can show a message box? Maybe the path is wrong.

**Reply by AntonioFS-1962** [reply]
Greetings from Sabadell-Barcelona-Spain.

Hello again.

I attach pdf with the steps followed according to instructions in GitHub. Also at the end I check that GTK3 is installed on my computer.

Thank you very much.
[20210926_VisualFBEditor.pdf](https://github.com/XusinboyBekchanov/VisualFBEditor/files/7231195/20210926_VisualFBEditor.pdf)

Antonio F.S.

**Reply by XusinboyBekchanov** [reply]
Fixed: Removed __USE_GTK3__ define from source (IDE):
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/87fa7fe5c478520764c7f8644070c0e7cac3ed6f

Fixed: Removed __USE_GTK3__ define from source (mff)
https://github.com/XusinboyBekchanov/MyFbFramework/commit/fa90ebaf5f0749b3bd4f286697e5f7bca2de45f2

Now try it, just compile this:
```
  cd Path_to_VisualFBEditor/src
  fbc "VisualFBEditor.bas" -x "../VisualFBEditor64_gtk3" -i "Path_to_VisualFBEditor/MyFbFramework" -d __USE_GTK3__
  cd Path_to_VisualFBEditor/MyFbFramework/mff
  fbc -b "mff.bi" -dll -x "../libmff64_gtk3.so" -d __USE_GTK3__
```

**Reply by XusinboyBekchanov** [reply]
Fixed: Compiling error in Linux:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/f3da691c56bf70e805a6a2f859e129cfd9a9713f

**Reply by AntonioFS-1962** [reply]
Hello.
Now I get the following error, after downloading the modified version you say:
![image](https://user-images.githubusercontent.com/91391996/135752344-22089906-58d1-4d86-8fbe-7c8172e28ca1.png)

**Reply by XusinboyBekchanov** [reply]
In my opinion, you have not downloaded the latest version of the library. What version of the FreeBasic compiler?

**Reply by AntonioFS-1962** [reply]
Hello.
The version of my FreeBasic is:
![image](https://user-images.githubusercontent.com/91391996/136451794-f363266b-0361-4e15-9c1c-c90a983b67aa.png)

**Reply by demosthenesk** [reply]
Make sure that you have downloaded both VisualFBEditor and MyFBFramework latest versions.
I compiled the latest of them in linux with no error.

**Reply by AntonioFS-1962** [reply]
https://user-images.githubusercontent.com/91391996/136699176-1878bf87-98c7-4ce7-8acc-2db29dd4139b.mov

https://user-images.githubusercontent.com/91391996/136699188-67e981ea-6c50-42e7-ae73-2be25263dfa3.mov

**Reply by XusinboyBekchanov** [reply]
I have updated the release version of MyFbFramework 1.2.9
Deleted definition `__USE_GTK3__`
Now there should be no mistake.

**Reply by AntonioFS-1962** [reply]
Hello again.

Good morning. Now I can start testing how the IDE works. Whenever I can (time constraints) I will report possible errors or malfunctions when coding with VisualFBEditor.

Best regards.
Antonio F.S.

![image](https://user-images.githubusercontent.com/91391996/136703775-43d0bf1a-ef80-4840-9f91-1104455491e0.png)

**Reply by XusinboyBekchanov** [reply]
This is the latest published version. You can see the new changes by downloading from Code -> Download ZIP:
![image](https://user-images.githubusercontent.com/32607344/136705186-517ce88c-b313-4447-9989-d64d5c55f0e9.png)

**Comment by XusinboyBekchanov** [comment]
https://user-images.githubusercontent.com/32607344/145813551-257b1b67-1550-4819-a219-5ec323e047a5.mp4
